src/impls.rs

```diff
@@ -195,6 +201,7 @@ pub fn register_key_impl(key: String, current_timestamp_sec: u64, pin: bool, wei
     update_key(&get_connection()?, key, peer_id, current_timestamp_sec.clone(), current_timestamp_sec, pin, weight)
 }
 
+/// used for replication, same as register_key, but key.pinned is ignored
```

Comment:
is it the same as `register_key`? Shouldn't it be different in that it avoids the `peer_id` check?

I'm not sure this check is OK for republishing. Wouldn't it prevent neighborhood from updating `timestamp_accessed`?
```rust
Ok(key) => key.peer_id == peer_id && key.timestamp_created < timestamp_created,
```

Also, I see it sets `timestamp_created` to `current_timestamp_sec`. Seems like a bug.

Reply:
Nope, it sets `timestamp_accessed` to `current_timestamp_sec` and other constraints checked in `update_key`

Reply:
oh, i see your point about `timestamp_created`

Reply:
In the case of republishing always should "win" the earliest key, I think. But we should discuss this on scrum tomorrow

Reply:
Yeah, makes sense.